scipy/special/_support_cupy_torch_jax.py

```diff
@@ -0,0 +1,50 @@
+import sys
+from scipy._lib.array_api_compat.array_api_compat import array_namespace
```

Comment:
`array_namespace` should be imported from `_lib._array_api.py`. This is a wrapper around the one from `array_api_compat`:
```
Thin wrapper around `array_api_compat.array_namespace`.

    1. Check for the global switch: SCIPY_ARRAY_API. This can also be accessed
       dynamically through ``_GLOBAL_CONFIG['SCIPY_ARRAY_API']``.
    2. `compliance_scipy` raise exceptions on known-bad subclasses. See
       it's definition for more details.
```

Reply:
Where should I import NumPy from? (as needed when array type is not recognized)

Reply:
Just `import numpy as np`, see my other comment.

Reply:
From https://github.com/scipy/scipy/pull/19023#discussion_r1296278554, I infer it's to be `import numpy as np`.

Reply:
Done.